### ISSUE TITLE
AWS EKS Kubeflow Katib Examples stuck in running (2/3 Ready)

### ISSUE THREAD

**Issue by charlescurt** [kubeflow/katib#1837]
/kind bug

**What steps did you take and what happened:**
I have been installing and uninstalling kubeflow multiple times now in different ways trying to get Katib to function on the basic examples. I am simply running the base examples, I tried multiple and all have the same issue. This most recent try was with the newer versions of k8 and kubeflow.

My error is something related to the worker pods being deployed as they all Error out at 2/3. I consistently get a terminated due to "Error." When looking into this error I have the below logs:

This issue seems closest to my issue: https://github.com/kubeflow/katib/issues/1258 I can't however figure out how to implement this fix on aws if it really is the issue. 

If you don't know the fix then perhaps some insight into what this issue actually is? I don't know what the katib controller is doing and what this certificate is.

My outputs from the commands: 
**kubectl port-forward svc/katib-controller -n kubeflow 8080:443**
**wget https://localhost:8080/mutate-experiments --no-check-certificate**
<img width="630" alt="Screen Shot 2022-03-21 at 4 46 00 PM" src="https://user-images.githubusercontent.com/98784768/159380917-31bca120-a0ec-489c-ab08-e38dd9b514b6.png">

`kubectl -n kubeflow-user-example-com logs random-wbddpcq4-9lfgl 
Using deprecated annotation `kubectl.kubernetes.io/default-logs-container` in pod/random-wbddpcq4-9lfgl. Please use `kubectl.kubernetes.io/default-container` instead
2022-03-22T18:26:00Z INFO     start with arguments Namespace(add_stn=False, batch_size=64, disp_batches=100, dtype='float32', gc_threshold=0.5, gc_type='none', gpus=None, image_shape='1, 28, 28', initializer='default', kv_store='device', load_epoch=None, loss='', lr=0.025957377119816605, lr_factor=0.1, lr_step_epochs='10', macrobatch_size=0, model_prefix=None, mom=0.9, monitor=0, network='mlp', num_classes=10, num_epochs=10, num_examples=60000, num_layers=4, optimizer='ftrl', profile_server_suffix='', profile_worker_suffix='', save_period=1, test_io=0, top_k=0, use_imagenet_data_augmentation=0, warmup_epochs=5, warmup_strategy='linear', wd=0.0001)
2022-03-22T18:26:00Z DEBUG    Starting new HTTP connection (1): data.mxnet.io:80
2022-03-22T18:26:00Z DEBUG    Starting new HTTP connection (1): data.mxnet.io:80
2022-03-22T18:26:00Z DEBUG    Starting new HTTP connection (1): data.mxnet.io:80
2022-03-22T18:26:00Z DEBUG    Starting new HTTP connection (1): data.mxnet.io:80
2022-03-22T18:26:00Z DEBUG    Starting new HTTP connection (1): data.mxnet.io:80
download failed, retrying, 4 attempts left
download failed, retrying, 3 attempts left
download failed, retrying, 2 attempts left
download failed, retrying, 1 attempt left
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/urllib3/connection.py", line 170, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw
  File "/usr/local/lib/python3.5/dist-packages/urllib3/util/connection.py", line 96, in create_connection
    raise err
  File "/usr/local/lib/python3.5/dist-packages/urllib3/util/connection.py", line 86, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py", line 706, in urlopen
    chunked=chunked,
  File "/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py", line 394, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.5/dist-packages/urllib3/connection.py", line 234, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/lib/python3.5/http/client.py", line 1151, in request
    self._send_request(method, url, body, headers)
  File "/usr/lib/python3.5/http/client.py", line 1196, in _send_request
    self.endheaders(body)
  File "/usr/lib/python3.5/http/client.py", line 1147, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python3.5/http/client.py", line 950, in _send_output
    self.send(msg)
  File "/usr/lib/python3.5/http/client.py", line 893, in send
    self.connect()
  File "/usr/local/lib/python3.5/dist-packages/urllib3/connection.py", line 200, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.5/dist-packages/urllib3/connection.py", line 182, in _new_conn
    self, "Failed to establish a new connection: %s" % e
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fcd15591978>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py", line 756, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/usr/local/lib/python3.5/dist-packages/urllib3/util/retry.py", line 573, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='data.mxnet.io', port=80): Max retries exceeded with url: /data/mnist/train-labels-idx1-ubyte.gz (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fcd15591978>: Failed to establish a new connection: [Errno 111] Connection refused',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/mxnet-mnist/mnist.py", line 86, in <module>
    fit.fit(args, sym, get_mnist_iter)
  File "/opt/mxnet-mnist/common/fit.py", line 185, in fit
    (train, val) = data_loader(args, kv)
  File "/opt/mxnet-mnist/mnist.py", line 44, in get_mnist_iter
    mnist = mx.test_utils.get_mnist()
  File "/usr/local/lib/python3.5/dist-packages/mxnet/test_utils.py", line 1907, in get_mnist
    path+'train-labels-idx1-ubyte.gz', path+'train-images-idx3-ubyte.gz')
  File "/usr/local/lib/python3.5/dist-packages/mxnet/test_utils.py", line 1894, in read_data
    with gzip.open(mx.test_utils.download(label_url)) as flbl:
  File "/usr/local/lib/python3.5/dist-packages/mxnet/test_utils.py", line 1812, in download
    raise e
  File "/usr/local/lib/python3.5/dist-packages/mxnet/test_utils.py", line 1802, in download
    r = requests.get(url, stream=True)
  File "/usr/local/lib/python3.5/dist-packages/requests/api.py", line 76, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.5/dist-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='data.mxnet.io', port=80): Max retries exceeded with url: /data/mnist/train-labels-idx1-ubyte.gz (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fcd15591978>: Failed to establish a new connection: [Errno 111] Connection refused',))`

**kubectl -n kubeflow-user-example-com get pods**
<img width="572" alt="Screen Shot 2022-03-21 at 4 20 43 PM" src="https://user-images.githubusercontent.com/98784768/159378677-7d170e4b-f283-4dc2-959b-551344b469af.png">

**kubectl -n kubeflow-user-example-com describe pods random-z2gxd48k-pdrwv | grep -B 3 -A 3 error**
<img width="298" alt="Screen Shot 2022-03-21 at 4 24 07 PM" src="https://user-images.githubusercontent.com/98784768/159378979-d8b8de97-a245-428c-8a0c-74eaeb04b227.png">

**kubectl -n kubeflow-user-example-com describe pods random-z2gxd48k-pdrwv | grep -B 3 -A 3 Error**
<img width="722" alt="Screen Shot 2022-03-21 at 4 25 00 PM" src="https://user-images.githubusercontent.com/98784768/159379051-e6dfbd91-022e-4568-8268-bf9c4a35e902.png">

**kubectl -n kubeflow-user-example-com get experiments**
<img width="229" alt="Screen Shot 2022-03-21 at 4 32 29 PM" src="https://user-images.githubusercontent.com/98784768/159379718-92082ef0-38af-4fe6-807a-075f1b02057b.png">

**kubectl -n kubeflow-user-example-com get trials**
<img width="272" alt="Screen Shot 2022-03-21 at 4 33 11 PM" src="https://user-images.githubusercontent.com/98784768/159379791-81310a67-0fb6-4f3f-905d-0d34fa54f44f.png">

**kubeops example**
<img width="388" alt="Screen Shot 2022-03-21 at 4 33 59 PM" src="https://user-images.githubusercontent.com/98784768/159379892-f0dea556-beff-4978-91a4-73408ed9cbf2.png">

**kubectl -n kubeflow-user-example-com logs random-z2gxd48k-pdrwv** 
<img width="729" alt="Screen Shot 2022-03-21 at 4 38 51 PM" src="https://user-images.githubusercontent.com/98784768/159380295-f3d2b57d-0b81-40ae-9019-9e03a207f829.png">

**What did you expect to happen:**
Examples to work following the standard documentation or at least to work when I updated them.

**Anything else you would like to add:**
Here are some links to issue I think are related:

I tried upgrading k8 because of this issue: https://github.com/istio/istio/issues/14389

Everything checks out from this issue: https://github.com/kubeflow/katib/issues/1160

**Environment:**

Enviroment: 
- AWS EKS
    -k8 1.21 
    - I have run 1-5 m5.xlarge instances with no difference depending on resources available

- kubeflow 1.4.1

- Katib: customize build https://github.com/awslabs/kubeflow-manifests/tree/main/docs/deployment/vanilla#central-dashboard)kustomize build apps/katib/upstream/installs/katib-with-kubeflow | kubectl apply -f -

- Kustomize: {Version:kustomize/v3.9.3 GitCommit:1ae8303bdc9372bc7c15942df6e9cf5d67fdba1a BuildDate:2021-02-07T17:02:13Z GoOs:linux GoArch:amd64}

Install Method:
- https://github.com/awslabs/kubeflow-manifests/tree/main/docs/deployment/vanilla

kubectl version:
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.5", GitCommit:"c285e781331a3785a7f436042c65c5641ce8a9e9", GitTreeState:"clean", BuildDate:"2022-03-16T15:58:47Z", GoVersion:"go1.17.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21+", GitVersion:"v1.21.5-eks-bc4871b", GitCommit:"5236faf39f1b7a7dabea8df12726f25608131aa9", GitTreeState:"clean", BuildDate:"2021-10-29T23:32:16Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.23) and server (1.21) exceeds the supported minor version skew of +/-1

---

Impacted by this bug? Give it a 👍 We prioritize the issues with the most 👍

**Comment by johnugeorge** [comment]
Can you try disabling istio for the katib job?

https://www.kubeflow.org/docs/components/katib/hyperparameter/#example-using-random-search-algorithm

**Comment by johnugeorge** [comment]
can you provide 

`kubectl get experiment -n kubeflow-user-example-com`

`kubectl get trials -n kubeflow-user-example-com`

`kubectl get pods -n kubeflow-user-example-com`

**Comment by charlescurt** [comment]
Sure!
<img width="574" alt="Screen Shot 2022-03-22 at 12 24 15 PM" src="https://user-images.githubusercontent.com/98784768/159559614-349cec20-979f-4703-8a75-95037508aede.png">

I also looked into the logs and saw this issue in master on both pytorch and example:
<img width="888" alt="Screen Shot 2022-03-22 at 12 24 58 PM" src="https://user-images.githubusercontent.com/98784768/159559794-d9bad1f6-fa94-48e6-9457-cb594fdd629a.png">

**Comment by charlescurt** [comment]
Here is the completed pods and master still running.
<img width="638" alt="Screen Shot 2022-03-22 at 12 31 15 PM" src="https://user-images.githubusercontent.com/98784768/159560798-5427221d-bec2-4153-9fe8-60d3107812eb.png">

**Comment by charlescurt** [comment]
PyTorch example addition:
<img width="567" alt="Screen Shot 2022-03-22 at 12 45 57 PM" src="https://user-images.githubusercontent.com/98784768/159563303-b255a111-3c21-480c-9873-f60095b35f94.png">

**Comment by johnugeorge** [comment]
Few things to be noted

Random example is not a distributed job. It is a just a batch job `Job`

Trials are not getting completed but corresponding jobs are completed. The likely reason is that metrics are not reported. 

I see that metrics collector container is missing in the training pod(Only 1 container as per your latest comment).However, in your earlier comment  https://github.com/kubeflow/katib/issues/1837#issue-1176096649, you reported that `metrics-logger-and-collector was present.

**Comment by johnugeorge** [comment]
Logs that you have provided does not report any error.  

To debug,

1. Please check why metrics collector was not created. 
2. Please check the status and logs of katib-mysql pod.

**Comment by charlescurt** [comment]
I appreciate your help. I am unsure what caused this issue. You are correct that the metrics collector was having problems, as was katib-mysql. I got a lot of errors around TCP timeouts. However, unlike the other comments of a similar situation, I allowed all traffic on my firewalls.

I did a complete reset, and everything started to work this time. I did fall back to k8's 1.19 when I remade the cluster.
